crates/parser/src/grammar/type_args.rs

```diff
@@ -65,17 +65,48 @@ fn generic_arg(p: &mut Parser) {
             m.complete(p, LIFETIME_ARG);
         }
         // test associated_type_bounds
-        // fn print_all<T: Iterator<Item: Display>>(printables: T) {}
-        IDENT if p.nth(1) == T![:] && p.nth(2) != T![:] => {
+        // fn print_all<T: Iterator<Item, Item::Item, Item: Display, Item<'a> = Item>>(printables: T) {}
+        IDENT if [T![<], T![=], T![:]].contains(&p.nth(1)) => {
+            let path_ty = p.start();
+            let path = p.start();
+            let path_seg = p.start();
             name_ref(p);
-            type_params::bounds(p);
-            m.complete(p, ASSOC_TYPE_ARG);
-        }
-        IDENT if p.nth(1) == T![=] => {
-            name_ref(p);
-            p.bump_any();
-            types::type_(p);
-            m.complete(p, ASSOC_TYPE_ARG);
+            if p.current() == T![<] {
```

Comment:
Looks like we're already in a `match p.current()` by this point, with its value currently bound to `IDENT`; would it be more intuitive to query against `IDENT` instead?

Reply:
The cursor position here has been moved due to the name_ref call prior to this.